components/learn/modules/ROOT/pages/write-contracts.adoc

```diff
@@ -0,0 +1,158 @@
+
+We can now write our first simple smart contract:
+
+```solidity
+pragma solidity ^0.5.0;
+
+contract SetAndGet {
+    uint256 private value;
+
+    // Emitted when the stored value changes
+    event ValueChanged(uint256 newValue);
+
+    // Stores a new value in the contract
+    function set(uint256 newValue) public {
+      value = newValue;
+
+      emit ValueChanged(newValue);
+    }
+
+    // Reads the last stored value
+    function get() public view returns (uint256) {
+      return value;
+    }
+}
+```
+
+We will store it in `contracts/StoreAndGet.sol`. Each `.sol` file should have the code for a single contract, and be named after it.
+
+[[compiling-solidity-source-code]]
+== Compiling Solidity Source Code
+
+The Ethereum Virtual Machine (EVM) cannot execute Solidity code directly: we first need to compile it into EVM bytecode.
+
+To do so, we'll use the https://docs.openzeppelin.com/sdk/2.6/[*OpenZeppelin command-line interface (CLI)*]:
+
+```bash
+$ npm install --save-dev @openzeppelin/cli
+```
+
+The CLI is quite powerful, and it is the entry point to many exciting features (such as upgradeable smart contracts and interactive transactions!). We will come back to it often.
+
+For now however, let's focus on compiling. This can be achieved by running a single command:
+
+```bash
+$ npx oz compile
+```
+
+NOTE: If you're unfamiliar with the `npx` command, check out our https://example.com[project setup guide].
+
+The `compile` command will automatically look for all contracts in the `contracts` directory, and compile them using the Solidity compiler version that best suits your project.
+
+You will notice a `build/contracts` directory was created: it holds the compiled artifacts (bytecode and metadata), which are `.json` files. It's a good idea to add this directory to your `.gitignore`.
+
+You can also configure your compilation, which includes choosing a compiler version and enabling optimizations, by passing arguments to the `compile` command:
+
+```bash
+$ npx oz compile --solc-version=0.5.12 --optimizer on
+```
+
+For detailed information on these options, refer to the https://docs.openzeppelin.com/sdk/2.6/api/cli#compile[`compile` API reference].
+
+[[multi-file-projects]]
+== Multi-File Projects
+
+There are multiple reasons that cause most projects to have multiple Solidity source files: from modularization via https://solidity.readthedocs.io/en/latest/contracts.html#inheritance[inheritance] to complex multi-contract systems.
+
+Because the compiler will pick up all files in the `contracts` directory and subdirectories, you are free to organize your code as you see fit (such as having subdirectories for `access-control` or `token` contracts).
+
+To import a contract from another file, refer to it by its relative path using an `import` statement:
+
+```solidity
+import './Math.sol'; // Import from the same directory
+import './access-control/Admin.sol'; // Import from a subdirectory
```

Comment:
WDYT about actually creating another contract, even if ridiculously simple, to import it? Otherwise it may be weird to see these two new names just spring up. We could write sth that uses the SetAndGet for instance!